base/lock.jl

```diff
@@ -51,6 +51,19 @@ end
 
 assert_havelock(l::ReentrantLock) = assert_havelock(l, l.locked_by)
 
+show(io::IO, ::ReentrantLock) = print(io, ReentrantLock, "()")
+
+function show(io::IO, ::MIME"text/plain", l::ReentrantLock)
+    show(io, l)
+    if !(get(io, :compact, false)::Bool)
+        if islocked(l)
+            print(io, " (locked by ", l.locked_by, ")")
```

Comment:
Let's add the word "current", and avoid a benign data-race that could print "(locked by nothing)" instead of "(unlocked)"
```suggestion
        locked_by = l.locked_by
        if locked_by isa Task
            print(io, " (locked by ", locked_by === current_task() ? "current " : "", locked_by, ")")
```